src/aiodynamo/expressions.py

```diff
@@ -0,0 +1,537 @@
+
+    def not_equals(self, other: Any) -> Condition:
+        return Comparison(F(self.name), "<>", other)
+
+
+class Parameters:
+    def __init__(self):
+        self.names: Dict[str, str] = {}
+        self.values: Dict[str, Dict[str, Any]] = {}
+        self.names_gen: Iterator[int] = count()
+        self.values_gen: Iterator[int] = count()
+        self.names_cache: _ParametersCache = {}
+        self.values_cache: _ParametersCache = {}
+
+    def encode_name(self, name: Union[str, int]) -> str:
+        return self._encode(
+            "#n", name, self.names, self.names_gen, self.names_cache, name
+        )
+
+    def encode_value(self, value: Any) -> Optional[str]:
+        value = low_level_serialize(value)
+        if value:
+            return self._encode(
+                ":v",
+                {value[0]: value[1]},
+                self.values,
+                self.values_gen,
+                self.values_cache,
+                value,
+            )
+        return None
+
+    def encode_path(self, path: KeyPath) -> str:
+        bits = [self.encode_name(path[0])]
+        for part in path[1:]:
+            if isinstance(part, int):
+                bits.append(f"[{part}]")
+            else:
+                bits.append(f".{self.encode_name(part)}")
+        return "".join(bits)
+
+    def get_expression_names(self) -> Dict[str, str]:
+        return self.names
+
+    def get_expression_values(self) -> Dict[str, Dict[str, Any]]:
+        return {key: value for key, value in self.values.items()}
+
+    def _encode(
+        self,
+        prefix: str,
+        thing: Any,
+        data: Dict[str, Any],
+        index_gen: Iterator[int],
+        cache: _ParametersCache,
+        cache_key: Any,
+    ) -> str:
+        try:
+            return cache[cache_key]
+
+        except KeyError:
+            can_cache = True
+        except TypeError:
+            can_cache = False
+        encoded = f"{prefix}{next(index_gen)}"
+        data[encoded] = thing
+        if can_cache:
+            cache[cache_key] = encoded
+        return encoded
+
+
+@dataclass(frozen=True)
+class HashAndRangeKeyCondition(KeyCondition):
+    hash_key: HashKey
+    range_key_condition: Condition
+
+    def encode(self, params: Parameters) -> str:
+        return f"{self.hash_key.encode(params)} AND {self.range_key_condition.encode(params)}"
+
+
+class Condition(metaclass=abc.ABCMeta):
+    def __and__(self, other: Condition) -> Condition:
+        return AndCondition(self, other)
+
+    def __or__(self, other: Condition) -> Condition:
+        return OrCondition(self, other)
+
+    def __invert__(self) -> Condition:
+        return NotCondition(self)
+
+    @abc.abstractmethod
+    def encode(self, params: Parameters) -> str:
+        pass
+
+
+@dataclass(frozen=True)
+class NotCondition(Condition):
+    base: Condition
+
+    def encode(self, params: Parameters) -> str:
+        return f"NOT {self.base.encode(params)}"
```

Comment:
Is it OK without parentheses?
As is does dynamo accept smth like `x OR NOT y` or does it demand `x OR (NOT y)` ?